WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductImageUploaderTests.swift

```diff
@@ -693,6 +729,265 @@ final class ProductImageUploaderTests: XCTestCase {
+
+        // Reset the notice flag
+        isNoticeTriggered = false
+
+        // Create an uploading status directly in storage
+        let uploadingStatus = ProductImageStatus.uploading(
+            asset: .uiImage(image: .checkmark, filename: "test", altText: "alt_test"),
+            siteID: siteID,
+            productID: productID
+        )
+        storage.addStatus(uploadingStatus)
+
+        // Wait for the active upload to be registered
+        waitUntil(timeout: 3) {
+            activeUploads.contains(key)
+        }
+
+        // When - With active uploads
+        imageUploader.sendBackgroundUploadNoticeIfNeeded(key: key, using: noticePresenter)
+
+        // Then
+        XCTAssertTrue(isNoticeTriggered, "Notice should be triggered when there are active uploads")
+    }
+
+    func test_reset_clears_storage_state_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+        let stores = MockStoresManager(sessionManager: .testingInstance)
+        let imageUploader = ProductImageUploader(stores: stores,
+                                                featureFlagService: mockFeatureFlagService)
+
+        let key = ProductImageUploaderKey(siteID: siteID,
+                                        productOrVariationID: productID,
+                                        isLocalID: false)
+
+        // Configure the store to keep uploads in progress. Don't call completion to keep it in uploading state.
+        stores.whenReceivingAction(ofType: MediaAction.self) { action in
+            if case .uploadMedia = action {
+            }
+        }
+
+        // Add a status to storage by uploading an image
+        let actionHandler = imageUploader.actionHandler(key: key, originalStatuses: [])
+        actionHandler.uploadMediaAssetToSiteMediaLibrary(asset: .phAsset(asset: PHAsset()))
+
+        // Wait for the upload to be registered
+        let uploadStatus = waitFor { promise in
+            actionHandler.addUpdateObserver(self) { statuses in
+                if statuses.hasPendingUpload {
+                    promise(statuses)
+                }
+            }
+        }
+
+        // Verify we have a pending upload
+        XCTAssertTrue(uploadStatus.hasPendingUpload)
+
+        // Set up subscription to track active uploads
+        var activeUploads: [ProductImageUploaderKey] = []
+        activeUploadsSubscription = imageUploader.activeUploads.sink { keys in
+            activeUploads = keys
+        }
+
+        // Wait for the active upload to be registered
+        waitUntil() {
+            activeUploads.contains { $0 == key }
+        }
+
+        // When
+        imageUploader.reset()
+
+        // Then - Active uploads should be cleared
+        waitUntil(timeout: 3) {
+            activeUploads.isEmpty
+        }
+    }
+
+    func test_replaceLocalID_is_called_when_flag_enabled() {
```

Comment:
Could you please explain why this test is necessary? From what I've seen, `replaceLocalID` method is not affected by the feature flag (no change in this PR)

Reply:
I read through the implementation of `replaceLocalID` again and still don't see any connection between this method and the status storage. It works solely with `actionHandlersByProduct`, an in-memory dictionary that holds the action handlers by product upload key. Do you think this implementation still works when the app is in the background?

If you see a connection between `replaceLocalID` and the status storage, please show me where the code that indicates that, because I cannot see it.

Reply:
This will not work when the app is in the background. It will only function when the app is in the foreground to maintain and properly store the current statuses. In the background, I need to use different logic for `replaceLocalID` because `ProductImageUploader` cannot operate while in the background. The idea is to use the same statuses when the app is in the background, fetching them from UserDefaults if there is anything pending. But this test is just to ensure that the status update replacing the local ID will work also with stored statuses (which means that the feature flag is enabled). If you believe it's useless to test that, I can remove it. In any case, at some point, it will remain only a test when the feature flag is removed.

Reply:
Since we have existing tests for `replaceLocalID` already, I think it's better to remove this test when it's unnecessary.